common/tqid/task_queue_validator.go

```diff
@@ -53,6 +79,15 @@ func NormalizeAndValidate(
 	taskQueue *taskqueue.TaskQueue,
 	defaultName string,
 	maxIDLengthLimit int,
+) error {
+	return normalizeAndValidate(taskQueue, defaultName, maxIDLengthLimit, false)
+}
+
+func normalizeAndValidate(
+	taskQueue *taskqueue.TaskQueue,
+	defaultName string,
+	maxIDLengthLimit int,
+	expectNonRootPartition bool,
```

Comment:
`expectNonRootPartition` feels slightly misleading since it's not "expect non-root", it's "expect root or non-root". So maybe invert it and call it `expectRoot`?